generic_server.cc

```diff
@@ -175,12 +173,11 @@ server::listen(socket_address addr, std::shared_ptr<seastar::tls::credentials_bu
 
 future<> server::do_accepts(int which, bool keepalive, socket_address server_addr) {
     return repeat([this, which, keepalive, server_addr] {
-        ++_connections_being_accepted;
-        return _listeners[which].accept().then_wrapped([this, keepalive, server_addr] (future<accept_result> f_cs_sa) mutable {
-            --_connections_being_accepted;
-            if (_stopping) {
+        seastar::gate::holder holder(_gate);
+        return _listeners[which].accept().then_wrapped([this, keepalive, server_addr, holder = std::move(holder)] (future<accept_result> f_cs_sa) mutable {
+            holder.release();
```

Comment:
Not really needed is it? You grab the holder inside the repeat, so it will be released on iteration return anyway.

Reply:
@elcallio I wanted to hold the gate as narrowly as possible; basically only for the duration of the `accept()`; just like the original code did. If I remove `holder.release()`, then the gate will only be released at the end of the lambda that is executed upon `accept()` completion (successful or exceptional).

... I do agree though that the rest of said lambda never blocks; the `(void)futurize_invoke(...).then_wrapped(...)` part only creates a chain of futures (which is not waited-for by anyone in particular, but will be processed asynchronously still). Thus dropping the explicit `holder.release()` should not make any difference because we won't switch to another task until the end of the scope anyway. I'll push v2.

Reply:
Done in v2.